frontend/src/homePage/CodeGroups.jsx

```diff
@@ -0,0 +1,124 @@
+import { useState, useEffect } from "react";
+import "./CodeGroups.css";
+import axios from "axios";
+
+function CodeGroups() {
+  const [allGroups, setAllGroups] = useState([]);
+  const [groupName, setGroupName] = useState("");
+  const [members, setMembers] = useState([]);
+  const [memberInput, setMemberInput] = useState("");
+  const [error, setError] = useState("");
+  const [count, setCount] = useState(0);
+
+  const BACKEND_URL = import.meta.env.VITE_BACKEND_URL;
+
+  useEffect(() => {
+    setAllGroups([]);
+    const token = localStorage.getItem("jwt");
+    if (!token) {
+      return;
+    }
+
+    const getGroups = async () => {
+      const options = {
+        headers: {
+          Authorization: `Bearer ${token}`,
+        },
+        withCredentials: true,
+      };
+
+      try {
+        const createdGroups = await axios.get(
+          `${BACKEND_URL}/api/user/created-groups`,
+          options
+        );
+
+        const addedGroups = await axios.get(
+          `${BACKEND_URL}/api/user/added-groups`,
+          options
+        );
+
+        setAllGroups([...createdGroups.data, ...addedGroups.data]);
+      } catch (error) {
+        setError("Error Fetching Groups.");
+      }
+    };
+    getGroups();
+  }, [count]);
+
+  const handleAddMembers = () => {
+    if (memberInput) {
+      setMembers([...members, memberInput]);
+      setMemberInput("");
+    }
+  };
+  const handleRemoveMember = (memberUsername) => {
+    setMembers(members.filter((member) => member !== memberUsername));
+  };
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+    try {
+      const response = await axios.post(
+        `${BACKEND_URL}/api/new/code-group`,
+        { groupName, members },
+        { withCredentials: true }
+      );
+      setCount((prev) => prev + 1);
```

Comment:
Similar to your [week 2 feedback](https://github.com/izuchukwumba/flixster/issues/16#:~:text=then%20you%20can%20simplify%20your%20code%20by%20calling%20the%20sortMovies/filterMovies%20directly%20in%20the%20button%20handler%20functions%2C%20instead%20of%20relying%20on%20a%20useEffect%20to%20call%20them.), Instead of triggering the useEffect by updating a count variable, I would directly call the API endpoint to fetch the groups.

Reply:
Fixed. Thanks